macOS/DuckDuckGo/Preferences/Model/SearchPreferences.swift

```diff
@@ -44,22 +43,25 @@ final class SearchPreferences: ObservableObject, PreferencesTabOpening {
     }
 
     init(persistor: SearchPreferencesPersistor = SearchPreferencesUserDefaultsPersistor(),
-         windowControllersManager: WindowControllersManager = Application.appDelegate.windowControllersManager) {
+         windowControllersManager: WindowControllersManagerProtocol) {
         self.persistor = persistor
         self.windowControllersManager = windowControllersManager
         showAutocompleteSuggestions = persistor.showAutocompleteSuggestions
     }
 
     private var persistor: SearchPreferencesPersistor
-    private var windowControllersManager: WindowControllersManager
+    let windowControllersManager: WindowControllersManagerProtocol
 
     @MainActor func openMoreSearchSettings() {
-        windowControllersManager.show(url: Self.moreSearchSettingsURL, source: .ui, newTab: true)
+        windowControllersManager.show(url: Self.moreSearchSettingsURL, source: .ui, newTab: true, selected: true)
     }
 }
 
 protocol PreferencesTabOpening {
 
+    var windowControllersManager: WindowControllersManagerProtocol { get }
```

Comment:
This change caused most of the refactoring. `windowControllersManager` is required for `openNewTab` so it must be required by the protocol, but a dozen of various classes and structs implement this protocol so they all had to be updated.